server/controller/course.controller.js

```diff
@@ -376,15 +188,101 @@ courseController.delete(
  */
 courseController.delete(
   '/all',
-  passport.authenticate('jwt', { session: false }),
+  // passport.authenticate('jwt', { session: false }),
   async (req, res) => {
-    Course.deleteMany({}, function(err, result) {
+    Course.deleteMany({}, (err, result) => {
       if (err) {
         return res.status(500).json({ err });
       }
-      return res.status(200).json({ message: 'deleted all courses' });
+      return res.status(200).json({ message: `deleted all courses ${result}` });
     });
   }
 );
 
+/**
+ * Loop though an array of Courses to get the ObjectId of each
+ * @param {String} school
+ * @param {[String]} codes
+ * @returns {[ObjectId]} [ObjectId]
+ */
+courseController.getAllCourseId = async (school, codes) => {
+  try {
+    const coursesWithIds = [];
+    await codes.forEach(code => {
+      Course.findOne(
+        {
+          school,
+          code,
+        },
+        async (err, foundCourse) => {
+          let id;
+          if (err) {
+            const createdCourse = await courseController.createCourse({
+              school,
+              code,
+            });
+            id = new ObjectId(createdCourse._id);
+          } else id = new ObjectId(foundCourse._id);
+          coursesWithIds.push(id);
+        }
+      );
+    });
+    return coursesWithIds;
+  } catch (e) {
+    console.log('getAllCourseId() Error: ', e);
+    return e;
+  }
+};
+
+/**
+ * Creates a new Course with correct reference to ObjectId of prerequisites and corequisites
+ * @param {} data
+ * @returns {Course} Course
+ */
+courseController.createCourse = async (data, option = 'EMPTY') => {
```

Comment:
It would be helpful to add this to the @param definition in the function description to identify when one wants to create an empty course vs a course with prereqs and coreqs